tests/flava/test_modeling_flava.py

```diff
@@ -0,0 +1,1234 @@
+                result.loss_info.mmm_text,
+                result.loss_info.mmm_image,
+            ]:
+                self.parent.assertEqual(item.dim(), 0)
+
+            for item in [result.loss_info.mim, result.loss_info.mlm]:
+                self.parent.assertIsNone(item)
+
+        else:
+            self.parent.assertIsNone(result.multimodal_masked_embeddings)
+            for item in [
+                result.loss_info.global_contrastive,
+                result.loss_info.itm,
+                result.loss_info.mmm_text,
+                result.loss_info.mmm_image,
+            ]:
+                self.parent.assertIsNone(item)
+
+        self.parent.assertIsNone(result.multimodal_embeddings)
+
+
+@require_torch
+class FLAVAForPreTrainingTest(FLAVAModelTest):
+    all_model_classes = (FLAVAForPreTraining,) if is_torch_available() else ()
+    class_for_tester = FLAVAForPreTrainingTester
+    test_torchscript = False
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    im = Image.open(requests.get(url, stream=True).raw)
+    return im
+
+
+@require_vision
+@require_torch
+class FLAVAModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference(self):
+        model_name = "aps/flava-full"
+        model = FLAVAModel.from_pretrained(model_name).to(torch_device)
+        processor = FLAVAProcessor.from_pretrained(model_name)
+
+        image = prepare_img()
+        inputs = processor(
+            text=["a photo of a cat", "a photo of a dog"],
+            images=[image, image],
+            padding="max_length",
+            max_length=77,
+            return_tensors="pt",
+            return_masks=False,
+        ).to(torch_device)
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs, return_dict=True)
+
+        # verify the embeddings
+        self.assertAlmostEqual(outputs.image_embeddings.sum().item(), -1352.53540, places=4)
+        self.assertAlmostEqual(outputs.text_embeddings.sum().item(), -198.98225, places=4)
+        self.assertAlmostEqual(outputs.multimodal_embeddings.sum().item(), -3988.51367, places=4)
+
+
+@require_vision
+@require_torch
+class FLAVAForPreTrainingIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference(self):
+        model_name = "aps/flava-full"
+        codebook_name = "aps/flava-codebook"
+        model = FLAVAForPreTraining.from_pretrained(model_name).to(torch_device)
+        codebook = FLAVACodebook.from_pretrained(codebook_name).to(torch_device)
+        codebook_fe = FLAVACodebookFeatureExtractor.from_pretrained(codebook_name)
+        processor = FLAVAProcessor.from_pretrained(model_name)
+        torch.manual_seed(1)
+        random.seed(1)
+
+        image = prepare_img()
+        mim_labels = codebook.get_codebook_indices(**codebook_fe([image, image], return_tensors="pt").to(torch_device))
+        inputs = processor(
```

Comment:
Think the `mlm_lables` should def be created outside of the processor (just like it's been done for the `mim_labels`)

Reply:
That sounds good to me but how to do that without using DataCollator?

Reply:
This has been resolved!